src/main/java/duke/storage/FileLoader.java

```diff
@@ -0,0 +1,102 @@
+package duke.storage;
+
+import duke.exceptions.DukeExceptionFileNotAccessible;
+import duke.exceptions.DukeExceptionFileNotWritable;
+import duke.exceptions.DukeExceptionIllegalArgument;
+import duke.tasks.TaskList;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class FileLoader {
+
+    protected File f;
+    public boolean isWritable;
+    public boolean isReadable;
```

Comment:
Should the variables be private? Getters can be written instead if necessary

Reply:
Yep